domain/consensus/model/externalapi/blockEqualClone_test.go

```diff
@@ -0,0 +1,334 @@
+package externalapi
+
+import (
+	"reflect"
+	"testing"
+)
+
+type blockToCompare struct {
+	block          *DomainBlock
+	expectedResult bool
+}
+
+type TestBlockStruct struct {
+	baseBlock         *DomainBlock
+	blocksToCompareTo []blockToCompare
+}
+
+func InitTestBaseTransactions() []*DomainTransaction {
+
+	testTx := []*DomainTransaction{{
+		Version:      1,
+		Inputs:       nil,
```

Comment:
Please don't use nil slices here and everywhere, since our `.Equal` doesn't differentiate between nil slices and empty slices, and `reflect.DeepEqual` does.